include/glow/Backend/BackendUtils.h

```diff
@@ -120,5 +120,36 @@ class RuntimeBundle {
 };
 } // namespace runtime
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH, const Function &F);
+
+/// If \p PH is an input placeholder, \returns true.
+bool isInput(const Placeholder *PH, const Function &F);
+
+/// If \p PH is an output placeholder in the function \p F, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH, const IRFunction &F);
+
+/// If \p PH is an input placeholder in the function \p F, \returns true.
+bool isInput(const Placeholder *PH, const IRFunction &F);
+
+/// Convert placeholders to be ordered as input|inputOutput|output|neither.
+/// Packed into {Placeholder *, isInput, isOutput}.
+template <typename FUN>
+std::vector<std::tuple<const Placeholder *, bool, bool>>
+getContiguousPlaceHolder(std::vector<const Placeholder *> const &holders,
+                         const FUN &F);
+
+/// An enum to indicate what type placholder it is.
+enum class PlaceholderType {
```

Comment:
This enum is used only by tests, right? If this is true, just move it into the CPP file with tests.

Reply:
> This enum is used only by tests, right? If this is true, just move it into the CPP file with tests.

OK. I moved them to the test cpp file.